semantics/c/language/translation/init.k

```diff
@@ -44,7 +48,89 @@ module C-TRANSLATION-INIT
           => UnknownCabsLoc
      rule filterCabsLoc(L:CabsLoc) => L [owise]
 
-     rule ExpressionLoc(K:KItem, _) => K
+     rule ExpressionLoc(K:K, L::CabsLoc) => ExpLoc(K, L)
+     rule typeof(ExpLoc(K:K, L::CabsLoc) => K)
+     rule utype(ExpLoc(K:K, L::CabsLoc) => K)
+     rule type(ExpLoc(K:K, L::CabsLoc) => K)
+
+     syntax KItem ::= #freezerExpLoc(CabsLoc)
+
+     rule
+          <k> ExpLoc(K:K, L::CabsLoc) => K ~> #freezerExpLoc(L) ... </k>
+          <elab>... .K => L </elab>
```

Comment:
This doesn't seem to be using the elab cell correctly. It's supposed to contain the elaborated result, but instead you're just appending a bunch of CabsLocs. That's fine for statements and declarations because they don't nest inside one another, but that doesn't work for expressions. What you want to be doing here is keeping the annotations unchanged and then dealing with the rest at execution time. It looks like you based this PR on the new C semantics, but it might have been better to model it based on the c++ semantics